src/scheduler-hackloop.js

```diff
@@ -49,35 +87,59 @@ export async function main(ns) {
       if (!meta) {
         ns.print(`WARNING: Could not find metadata for server: ${target}`);
       } else {
+        // add faction script to meta for faction servers
+        if (factionServers.includes(target)) {
+          meta.runScript = 'backdoor.js';
+        }
         // Check if a script is running
-        const isRunning = ns.isRunning(hackScript, target);
+        const isRunningLocal = ns.isRunning(hackScriptLocal, currentHost, target); // run hackloop.js target
+        const isRunningRemote = ns.isRunning(hackScriptRemote, target); // remote exec, see below
 
         // Check if the server is hacked and we could hack it
         const hackSkillMet = hackingSkill >= meta.reqHackSkill;
         const toolCountMet = hackToolCount >= meta.reqNukePorts;
         const alreadyHacked = ns.hasRootAccess(target);
+
         const shouldRun =
           hackSkillMet &&
           toolCountMet &&
-          !isRunning &&
           (meta.restartHack || !alreadyHacked);
+        const shouldRunLocal =
+          shouldRun &&
+          !isRunningLocal &&
+          (runType === 'local' || runType === 'both');
+        const shouldRunRemote =
+          shouldRun &&
+          !isRunningRemote &&
+          (runType === 'remote' || runType === 'both');
 
-        if (shouldRun) {
-          ns.print(`Starting hack on ${target}`);
-          // ns.exec(meta.runScript || 'hackloop.js', 'home', 1, target);
+        if (shouldRunLocal) {
+          ns.toast(`Starting hack locally against ${target}`);
+          ns.print(`Starting hack locally against ${target}`);
+          ns.exec(meta.runScript || hackScriptLocal, currentHost, 1, target);
+        }
+        if (shouldRunRemote) {
+          ns.toast(`Starting hack remotely on ${target}`);
           const remotePid = ns.exec(
-            'prepServer.js',
-            'home',
+            crackScript,
+            currentHost,
             1,
             target,
-            'hackloop-remote.js',
+            hackScriptRemote,
           );
-          ns.print(`Spawning on remote ${target} pid ${remotePid}`);
-          await ns.sleep(1000);
+          if (remotePid === 0) {
+            ns.toast(`FAILED to run crackScript ${crackScript} on remote ${target}`);
+            ns.print(`FAILED to run crackScript ${crackScript} on remote ${target}`);
+          } else {
+            ns.toast(`Spawning on remote ${target} pid ${remotePid}`);
+            ns.print(`Spawning on remote ${target} pid ${remotePid}`);
+          }
         }
+        await ns.sleep(500); // sleep 0.5 second to give time for next server on this pass
       }
     }
 
     await ns.sleep(sleepTime);
+    ns.print(`Snoozing for ${sleepTime/1000} seconds.`)
```

Comment:
FWIW I believe just `ns.print(``Snoozing for ${ns.nFormat(sleepTime)}.``);` (substituting the double backtick for a single one b/c of GH MD formatting...